src/sentry/runner/initializer.py

```diff
@@ -201,13 +201,19 @@ def apply_legacy_settings(settings):
                       "Use SENTRY_OPTIONS instead, key 'system.admin-email'", DeprecationWarning)
         settings.SENTRY_OPTIONS['system.admin-email'] = settings.SENTRY_ADMIN_EMAIL
 
-    if settings.SENTRY_URL_PREFIX in ('', 'http://sentry.example.com') and not settings.DEBUG:
-        # Maybe also point to a piece of documentation for more information?
-        # This directly coincides with users getting the awkward
-        # `ALLOWED_HOSTS` exception.
-        show_big_error('SENTRY_URL_PREFIX is not configured')
-        # Set `ALLOWED_HOSTS` to the catch-all so it works
-        settings.ALLOWED_HOSTS = ['*']
+    if not settings.SENTRY_OPTIONS.get('system.url-prefix') and hasattr(settings, 'SENTRY_URL_PREFIX'):
+        import warnings
+        warnings.warn('SENTRY_URL_PREFIX is deprecated.'
+                      "Use SENTRY_OPTIONS instead, key 'system.url-prefix'", DeprecationWarning)
+        settings.SENTRY_OPTIONS['system.url-prefix'] = settings.SENTRY_URL_PREFIX
+
+    if not hasattr(settings, 'SENTRY_URL_PREFIX'):
+        from sentry import options
+        url_prefix = options.get('system.url-prefix', silent=True)
+        if not url_prefix:
+            # HACK: We need to have some value here for backwards compatibility
+            url_prefix = 'http://sentry.example.com'
+        settings.SENTRY_URL_PREFIX = url_prefix
```

Comment:
So this one needs to be done here since I want to fetch the value from the database and use that if we have an override.

This is a special case, and should probably just go away soon. I wouldn't want to prefill `SENTRY_URL_PREFIX` with the value coming from disk since that was unlikely overridden.